specs/lightclient/validator.md

```diff
@@ -0,0 +1,198 @@
+
+HF 1 introduces a new type of committee: the sync committee. Sync committees are responsible for signing each block of the canonical chain and there exists an efficient algorithm for light clients to sync the chain using the output of the sync committees. See the [sync protocol](./sync-protocol.md) for further details on the light client sync. Under this network upgrade, validators will need to track their participation in this new committee type and produce the relevant signatures as required. Block proposers will need to incorporate the (aggregated) sync committee signatures into each block they produce.
+
+## Prerequisites
+
+All terminology, constants, functions, and protocol mechanics defined in the [HF 1 -- The Beacon Chain](./beacon-chain.md) doc are requisite for this document and used throughout. Please see this document before continuing and use as a reference throughout.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SyncCommitteeIndex` | `uint64` | Index of a validator within a sync committee |
+
+## Containers
+
+### SyncCommitteeContribution
+
+```python
+class SyncCommitteeContribution(Container):
+    # Slot during which this contribution is broadcast
+    slot: Slot
+    # Position of the validator within the sync committee
+    index: SyncCommitteeIndex
+    # Signature by the validator over the block root of the _previous_ slot
+    signature: BLSSignature
+```
+
+## Validator assignments
+
+A validator determines attestation committee assignments and beacon block proposal duties as defined in the Phase 0 document.
+
+To determine sync committee assignments, a validator can run the following function: `get_sync_committee_assignment(state, epoch, validator_index)` where `epoch` is an epoch number within the current or next sync committee period. This function will return a tuple of `SyncCommitteeIndex` for each position in the sync committee that the pubkey corresponding to `validator_index` appears. If the validator has no assignments in the corresponding sync committees, this tuple will be empty `()`. 
+
+*Note*: `SyncCommitteeIndex` is different than the index into the validator registry or some attestation committee.
+
+*Note*:  A single validator may be duplicated in a given sync committee.
+
+```python
+def compute_sync_committee_period(epoch: Epoch) -> uint64:
+    return epoch // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+
+def indices_in_sync_committee(sync_committee: SyncCommittee, target_pubkey: BLSPubkey) -> Tuple[SyncCommitteeIndex, ...]:
+    return tuple(SyncCommitteeIndex(index) for index, pubkey in enumerate(sync_committee.pubkeys) if pubkey == target_pubkey)
+
+def get_sync_committee_assignment(state: BeaconState,
```

Comment:
If we remove `SyncCommitteeIndex` and instead put this on the aggregator to include duplicates, this can just return a boolean. `return pubkey in state.next_sync_committee`